### PR TITLE
enable login.gov on staging

### PR DESCRIPTION
**Pull request by terrazoon**

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -60,6 +60,13 @@ jobs:
           NEW_RELIC_LICENSE_KEY: ${{ secrets.NEW_RELIC_LICENSE_KEY }}
           NR_BROWSER_KEY: ${{ secrets.NR_BROWSER_KEY }}
           LOGIN_PEM: ${{ secrets.LOGIN_PEM }}
+          # Need to double check value of client id with Carlo in LOGIN_DOT_GOV_CLIENT_ID and LOGIN_DOT_GOV_LOGOUT_URL
+          LOGIN_DOT_GOV_CLIENT_ID: "urn:gov:gsa:openidconnect.profiles:sp:sso:gsa:notify_gov"
```
Comment (ccostino) [inline .github/workflows/deploy.yml:64]: Good thing on double checking!  It's with a dash, not an underscore. :-)

```suggestion
          LOGIN_DOT_GOV_CLIENT_ID: "urn:gov:gsa:openidconnect.profiles:sp:sso:gsa:notify-gov"
```

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -60,6 +60,13 @@ jobs:
...
+          LOGIN_DOT_GOV_USER_INFO_URL: "https://idp.int.identitysandbox.gov/api/openid_connect/userinfo"
+          LOGIN_DOT_GOV_ACCESS_TOKEN_URL: "https://idp.int.identitysandbox.gov/api/openid_connect/token"
+          LOGIN_DOT_GOV_LOGOUT_URL: "https://idp.int.identitysandbox.gov/openid_connect/logout?client_id=urn:gov:gsa:openidconnect.profiles:sp:sso:gsa:notify_gov&post_logout_redirect_uri=https://notify-staging.app.cloud.gov/sign-out"
```
Comment (ccostino) [inline .github/workflows/deploy.yml:67]: Same here:

```suggestion
          LOGIN_DOT_GOV_LOGOUT_URL: "https://idp.int.identitysandbox.gov/openid_connect/logout?client_id=urn:gov:gsa:openidconnect.profiles:sp:sso:gsa:notify-gov&post_logout_redirect_uri=https://notify-staging.app.cloud.gov/sign-out"
```

File: app/main/views/sign_in.py [hunk 3]
```diff
@@ -153,12 +153,34 @@ def sign_in():
...
+        f"LOGIN_DOT_GOV_SIGNOUT_REDIRECT={os.getenv('LOGIN_DOT_GOV_SIGNOUT_REDIRECT')}"
+    )
+
     return render_template(
         "views/signin.html",
         form=form,
         again=bool(redirect_url),
         other_device=other_device,
-        notify_env_is_dev=bool(notify_env == "development"),
+        notify_env_is_dev_or_staging=bool(notify_env in ["development", "staging"]),
```
Comment (ccostino) [inline app/main/views/sign_in.py:183]: Would it make sense to change this variable to something like `login_gov_enabled` instead?

```suggestion
        login_gov_enabled=bool(notify_env in ["development", "staging"]),
```

File: app/templates/views/signin.html [hunk 4]
```diff
@@ -28,7 +28,7 @@ <h1 class="font-body-2xl margin-bottom-3">You need to sign in again</h1>
       {% endif %}
     {% else %}
       <h1 class="font-body-2xl margin-bottom-3">Sign in</h1>
-      {% if notify_env_is_dev %}
+      {% if notify_env_is_dev_or_staging %}
```
Comment (ccostino) [inline app/templates/views/signin.html:31]: And then here as well?

```suggestion
      {% if login_gov_enabled %}
```

**Review by ccostino — COMMENTED** [review]
Thank you, @terrazoon!  I left a few comments for adjusting a couple of things like the URN and the variable; once those are in then we'll merge and give it a go!

File: app/main/views/sign_in.py (hunk 3 above)
Comment (terrazoon) [reply app/main/views/sign_in.py:183]: I guess so.  The variable is going away once login.gov is available on prod, so it's temporary anyway.

File: app/templates/views/signin.html (hunk 4 above)
Comment (terrazoon) [reply app/templates/views/signin.html:31]: I suppose.  It's going to be replaced anyway.

**Review by ccostino — APPROVED** [review]
Thanks, @terrazoon!  Off we go! 🚀

**Pull request merged**